src/OrleansCodeGenerator/GrainReferenceGenerator.cs

```diff
@@ -191,7 +192,8 @@ private static MemberDeclarationSyntax[] GenerateInvokeMethods(Type grainType, A
                 var options = GetInvokeOptions(method);
 
                 // Construct the invocation call.
-                if (method.ReturnType == typeof(void))
+                var isOneWayTask = method.GetCustomAttribute<OneWayAttribute>() != null;
+                if (method.ReturnType == typeof(void) || isOneWayTask)
```

Comment:
I was going to suggest to also support void returning methods only if they are marked OneWay, so that if the grain implementation doesn't do any asynchronous work, that both the caller and implementation benefit of not involving tasks in their usage.
I assume this is not enough and codegen would need to be updated to allow that path, correct?